src/hotspot/share/jfr/recorder/stacktrace/jfrStackTraceRepository.hpp

```diff
@@ -52,6 +52,8 @@ class JfrStackTraceRepository : public JfrCHeapObj {
   JfrStackTraceRepository();
   static JfrStackTraceRepository* create();
   static void destroy();
+  static JfrStackTraceRepository* create_leak_profiler();
+  static void destroy_leak_profiler();
```

Comment:
Does it make sense to have 2 distinct `destroy()` methods here?
Couldn't the original `destroy()` take care of both the ordinary and leakp instance?

Reply:
It is possible to combine both `destroy()` but I think there is an underlying convention to have 1 object - 1 constructor - 1 destructor. It seems to be the case everywhere in JFR code.

Reply:
👍